libcperciva/util/sock.c

```diff
@@ -432,6 +432,48 @@ sock_connect_nb(const struct sock_addr * sa)
 	return (-1);
 }
 
+/**
+ * sock_connect_bind_nb(sa, sa_b):
+ * Create a socket, mark it as non-blocking, and attempt to connect to the
+ * address ${sa} after binding to ${sa_b}.  Return the socket (connected
+ * or in the process of connecting) or -1 on error.
+ */
+int
+sock_connect_bind_nb(const struct sock_addr * sa, const struct sock_addr * sa_b)
+{
+	int s;
+
+	/* Create a socket. */
+	if ((s = socket(sa->ai_family, sa->ai_socktype, 0)) == -1)
+		goto err0;
+
+	/* Bind the socket to sa_b. */
+	if ((bind(s, sa_b->name, sa_b->namelen)) == -1)
+		warnp("Error binding socket");
+		goto err1;
+
```

Comment:
Missing brackets!

This always does goto err1, regardless of whether there's an error or not.

(I previously highlighted the top of this problematic section, but github shows 4 lines above the indicated line, so that didn't show the problem as well as highlighting this line.)

Reply:
Oops, I took it from other part of the source and missed brackets, sorry. Fixed.